shared/macros-oval.jinja

```diff
@@ -0,0 +1,64 @@
+{{%- macro oval_lineinfile(msg='', path='', parameter='', value='', default='', application='', rpm_package='') -%}}
```

Comment:
I believe that the macro `oval_lineinfile` should be more basic, just expand into a `textfilecontent54_test`, `textfilecontent54_object`, and optionally a `textfilecontent54_state`.

Reply:
The `rpm_package` parameter seems out of place in the macro.

Reply:
I largely agree with @yuumasato here.

More generally, I think the OVAL should condition on the status of the file (Does `/etc/ssh/sshd_config` exist at all? -- and conditionally fail based on a jinja-parameter like `missing_fail`?). We'd then make `missing_fail` condition on the status of the RPM package: if `sshd` isn't installed, a missing `/etc/ssh/sshd_config` is fine. If `sshd` is installed, a missing `/etc/ssh/sshd_config` is bad, because that means we don't know where the config for it resides. That seems to be the more robust solution. 

The other problem we have with OVAL is that we have to put everything inside a single `<def-group />` element, but some portions need to be inside the `criteria` sub-element while others can be appended to the end. This makes it hard to define a single `lineinfile`-type macro for it. I think we should have a way of expanding different portions (`criteria`, `test`, `object`, and `state`), and then build higher-level macros over these as we need to.

Reply:
Agree. The best it would be to have macros for basic things - `test`, `object` and `state`. These basic macros could be used in macro for one check and the "one check" macros in whole rule check (e.g. `sshd`).